src/Lecture3.hs

```diff
@@ -210,8 +235,15 @@ types that can have such an instance.
 -- instance Foldable Weekday where
 -- instance Foldable Gold where
 -- instance Foldable Reward where
--- instance Foldable List1 where
--- instance Foldable Treasure where
+
+instance Foldable List1 where
+  foldr f z (List1 a []) = f a z
+  foldr f z (List1 _ (x : xs))  = f x (foldr f z xs)
```

Comment:
It looks like this instance can be simplified by delegating the `foldr` implementation to ordinary lists
```suggestion
  foldr f z (List1 x xs) = foldr f z (x : xs)
```